### PR TITLE
Include all segments that have ANY content in the trim filter range (…

### PR DESCRIPTION
**Pull request by keithrichardscbs**
…#107)

### CODE REVIEW

File: docs/filters/trim.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ nav_order: 4
 ---
 
 # Trim
-An **INCLUSIVE RANGE** of segments to **INCLUDE** in the modified variant playlist. Segments outside this range will be filtered out. The Playlist returned will be a Video on Demand Playlist. 
+An **INCLUSIVE RANGE** of segments to **INCLUDE** in the modified variant playlist. Segments that contain ANY amount of content included in the range will be included. Segments with NO content in the range will be filtered out. The Playlist returned will be a Video on Demand Playlist.
```
Comment (krystalmejia24) [inline docs/filters/trim.md:8]: THANKS FOR UPDATING THE DOCS! :)

File: filters/hls.go [hunk 2]
```diff
@@ -372,7 +379,15 @@ func (h *HLSFilter) filterRenditionManifest(filters *parsers.MediaFilters, m *m3
...
-		append = inRange(filters.Trim.Start, filters.Trim.End, int(segment.ProgramDateTime.Unix()))
+		segmentTimestamp := int(segment.ProgramDateTime.Unix())
+
+		isInRange := inRange(filters.Trim.Start, filters.Trim.End, segmentTimestamp)
+		if shouldIncludePreviousSegmentInTrimmedRenditionManifest(append, isInRange, i, segmentTimestamp, filters.Trim.Start) {
+			if err := appendSegment(h.manifestURL, m.Segments[i-1], filteredPlaylist); err != nil {
+				return "", fmt.Errorf("trimming segments: %w", err)
+			}
+		}
+		append = isInRange
```
Comment (krystalmejia24) [inline filters/hls.go:390]: one thing to keep in mind is that the segments are held in a circular buffer so lets always check if the previous segment is nil. https://github.com/grafov/m3u8/issues/127 

Also, im wondering if we can just update the `append` mode by checking the current segment end? 
```
	segmentTimestamp := int(segment.ProgramDateTime.Unix())
	if append = inRange(filters.Trim.Start, filters.Trim.End, segmentTimestamp); !append {
		currentSegmentEnd := segmentTimestamp + int(segment.Duration)
		append = inRange(filters.Trim.Start, filters.Trim.End, currentSegmentEnd)
	}
```

less dependencies on index, nil segments.

Not sure if yout think we need to introduce `inclusiveInRange` and `exclusiveInRange` im assuming the current `InRange` function, which is inclusive, is enough for checking the end time of the segment.

File: filters/hls_test.go [hunk 3]
```diff
@@ -1669,6 +1669,31 @@ https://existing.base/path/chan_1/chan_1_20200311T202813_1_00024.ts
 #EXTINF:6.000,
 https://existing.base/path/chan_1/chan_1_20200311T202818_1_00025.ts
 #EXT-X-ENDLIST
+`
```
Comment (krystalmejia24) [inline filters/hls_test.go:1672]: lets also add tests here https://github.com/cbsinteractive/bakery/blob/6137a3f8a9bac78a29b83bf43a61610a67f6478c/filters/hls_test.go#L1784

when adding tests cases to this it would be good to really exercise those boundaries. Something like having the ad tags, which have no PDT, sit right after a segment whose exclusively in range.

File: filters/hls.go (hunk 2 above)
Comment (keithrichardscbs) [reply filters/hls.go:390]: Yeah I like this. However, we will have to keep the check against the start time because if the segment ends right at the start of the range, inRange would return true (and we want to exlude it in that case).

**Comment by keithrichardscbs** [comment]
@krystalmejia24 I think you may have force pushed over the changes you requested. Suddenly the commits don't match my local branch.

**Comment by krystalmejia24** [comment]
@keithrichardscbs i totally did do that! I accidentally rebased on the wrong commit as I had pulled your changes in previously, but never pulled in the newest commit with change we discussed. 

I have cleaned that up https://github.com/cbsinteractive/bakery/pull/109/files

as you can see its just your two changes and the rebase i made against master branch. Im pulling your changes into that branch as they are all related. Once i can evaluate both working with eachother will merge to dev!
